src/platform/Linux/NetworkCommissioningWiFiDriver.cpp

```diff
@@ -34,16 +34,63 @@ namespace DeviceLayer {
 namespace NetworkCommissioning {
 
 #if CHIP_DEVICE_CONFIG_ENABLE_WPA
+// The length value is the strlen of the keyname plus the number of networks allowed followed by a null terminator.
+// Example: "wifi-ssid" has a string len of 9. If the max network supported is 2 then we want to create keys
+// "wifi-ssid0" and "wifi-ssid1". With the below method we can only have a max of 10 networks as the lengths would
+// change if wanting to have wifi-ssid10.
+#define SSID_KEY_NAME_LEN 11 // wifi-ssid
+#define PASS_KEY_NAME_LEN 11 // wifi-pass
+#if CHIP_DEVICE_CONFIG_ENABLE_WIFI_PDC
+#define NETWORK_ID_KEY_NAME_LEN 9     // wifi-ni
+#define CLIENT_ID_KEY_NAME_LEN 9      // wifi-ci
+#define CLIENT_ID_KEYPAIR_NAME_LEN 10 // wifi-cik
+#endif                                // CHIP_DEVICE_CONFIG_ENABLE_WIFI_PDC
+
 // TODO(#14172): Here, most interfaces are just calling ConnectivityManager interfaces, this is because the ConnectivityProvides
 // some bootstrap code for the wpa_supplicant. However, we can wrap the wpa_supplicant dbus api directly (and remove the related
 // code in ConnectivityManagerImpl).
 namespace {
-constexpr char kWiFiSSIDKeyName[]        = "wifi-ssid";
-constexpr char kWiFiCredentialsKeyName[] = "wifi-pass";
+constexpr char intToChar(int i)
+{
+    char c = i + '0';
+    return c;
+}
+
+template <size_t size>
+constexpr std::array<char[size], kMaxNetworks> createNetworkStorageKeys(const char keyName[size])
+{
+    std::array<char[size], kMaxNetworks> arr{};
+
+    for (int i = 0; i < kMaxNetworks; i++)
+    {
+        for (size_t j = 0; j < size; j++)
+        {
+            arr[i][j] = keyName[j];
+        }
+        arr[i][size - 2] = intToChar(i);
```

Comment:
So this is assuming that keyName[size - 1] is already 0?  Can we static_assert that, or is that not consteval (and if so, in what sense is any of this constexpr)?

Reply:
Yes the assumption is that the keyName[size - 1] is 0 or null terminated. We can add a static assert but when adding the static assert in the function the function is no longer a consteval. However we could add a static assert for every individual keyname before calling `createNetworkStorageKeys` . Eg. 
`constexpr char ssidKeyName[SSID_KEY_NAME_LEN] = "wifi-ssid";
static_assert(ssidKeyName[SSID_KEY_NAME_LEN - 1] == 0, "ssidKeyName not null terminated");`
Thoughts on having it a static assert for every key name?

Reply:
Adding asserts for every key name does not really make sense, in that it tells us nothing about what's seen inside the function that makes the assumption.

I guess we can just document the assumption, and make this function consteval once we switch to c++20....